File: LeetCode/Basic/1608/main.go
package main

func specialArray(nums []int) int {
	var cnt, max int
	max = -1
	for i := 0; i <= len(nums); i++ {
		cnt = 0
		for _, num := range nums {
			if num >= i {
				cnt++
			}
		}
		if i == cnt && max < i {
			max = i
		}
	}
	return max
}

File: LeetCode/Basic/1608/main.py
from typing import List


class Solution:
    def specialArray(self, nums: List[int]) -> int:
        max = -1
        for idx in range(0, len(nums)+1):
            cnt = 0
            for num in nums:
                if num >= idx:
                    cnt += 1
            if idx == cnt and max < cnt:
                max = cnt
        return max
